src/overlay/Floodgate.cpp

```diff
@@ -58,7 +58,7 @@ bool
 Floodgate::addRecord(StellarMessage const& msg, Peer::pointer peer, Hash& index)
 {
     ZoneScoped;
-    index = sha256(xdr::xdr_to_opaque(msg));
+    index = xdrBlake2(msg);
```

Comment:
you forgot to update `Tracker::listen` (so this commit breaks item fetcher)

Reply:
I .. don't think so! The item fetcher and tracker deals in public sha256 hashes -- the shared identities of txsets and qsets baked into the public protocol. We're not changing its hashes.

The floodmap is an entirely internal usage, the hashes that key it never leave it. They're private book-keeping, which is why it's safe to change them here (and changing them is unrelated to the hashes used elsewhere).

Reply:
no: look at how `getPeersKnows` (which now uses Blake2 hashes) is used: it's passing what was computed in `Tracker::listen`, which is computed as:

```
    // NB: hash here is of StellarMessage
    mWaitingEnvelopes.push_back(
        std::make_pair(sha256(xdr::xdr_to_opaque(m)), env));
```

Those keys are only used in connection to the flood map from what I can tell. The entire code base, except for the actual download code only uses the values.

This should not be confused with *what* item fetcher is trying to download (which is also a hash but of the item that we're trying to get).

Reply:
Oh my goodness. Nice catch! Fixed (and added a comment).